cluster-controller/src/main/java/io/strimzi/controller/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -0,0 +1,250 @@
+        List<Future> result = new ArrayList<>(4 + (deleteClaims ? zk.getReplicas() : 0));
+
+        result.add(configMapOperations.reconcile(namespace, ZookeeperCluster.zookeeperMetricsName(name), null));
+        result.add(serviceOperations.reconcile(namespace, ZookeeperCluster.zookeeperClusterName(name), null));
+        result.add(serviceOperations.reconcile(namespace, ZookeeperCluster.zookeeperHeadlessName(name), null));
+        result.add(zkSetOperations.reconcile(namespace, ZookeeperCluster.zookeeperClusterName(name), null));
+
+        if (deleteClaims) {
+            for (int i = 0; i < zk.getReplicas(); i++) {
+                result.add(pvcOperations.reconcile(namespace, zk.getPersistentVolumeClaimName(i), null));
+            }
+        }
+
+        // TODO wait for endpoints and pods to disappear
+
+        return CompositeFuture.join(result);
+    };
+
+    private final Future<ReconcileResult<Void>> createOrUpdateTopicController(Reconciliation reconciliation, ConfigMap assemblyCm) {
+        String namespace = assemblyCm.getMetadata().getNamespace();
+        String name = assemblyCm.getMetadata().getName();
+        log.info("{}: create/update topic controller {}", reconciliation, name);
+        TopicController topicController = TopicController.fromConfigMap(assemblyCm);
+        Deployment deployment = topicController != null ? topicController.generateDeployment() : null;
+        return deploymentOperations.reconcile(namespace, topicControllerName(name), deployment);
+    };
+
+    private final Future<ReconcileResult<Void>> deleteTopicController(Reconciliation reconciliation) {
+        String namespace = reconciliation.namespace();
+        String name = reconciliation.assemblyName();
+        log.info("{}: delete topic controller {}", reconciliation, name);
+        return deploymentOperations.reconcile(namespace, topicControllerName(name), null);
+        // TODO wait for pod to disappear
+    };
+
+    @Override
+    protected void delete(Reconciliation reconciliation, Handler<AsyncResult<Void>> handler) {
+        Future<Void> f = Future.<Void>future().setHandler(handler);
+        deleteTopicController(reconciliation)
+                .compose(i -> deleteKafka(reconciliation))
+                .compose(i -> deleteZk(reconciliation))
+                .compose(ar -> f.complete(), f);
+    }
+
+    @Override
+    protected List<HasMetadata> getResources(String namespace, Labels selector) {
+        List<HasMetadata> result = new ArrayList<>();
+        result.addAll(kafkaSetOperations.list(namespace, Labels.forType(KafkaCluster.TYPE)));
+        result.addAll(zkSetOperations.list(namespace, Labels.forType(KafkaCluster.TYPE)));
+        result.addAll(deploymentOperations.list(namespace, Labels.forType(KafkaCluster.TYPE)));
+        result.addAll(serviceOperations.list(namespace, Labels.forType(KafkaCluster.TYPE)));
+        // TODO remove the cluster cm! result.addAll(configMapOperations.list(namespace, selector));
```

Comment:
I'm not sure what exactly was meant with this TODO. But should we also look for the metrics config maps here? (that could be easily a minor for another PR)